src_python/habitat_sim/utils/settings.py

```diff
@@ -132,62 +70,71 @@ def create_fisheye_spec(**kw_args):
         fisheye_sensor_spec.alpha = 0.57
         fisheye_sensor_spec.focal_length = [364.84, 364.86]
 
-        fisheye_sensor_spec.resolution = [settings["height"], settings["width"]]
         # The default principal_point_offset is the middle of the image
         fisheye_sensor_spec.principal_point_offset = None
         # default: fisheye_sensor_spec.principal_point_offset = [i/2 for i in fisheye_sensor_spec.resolution]
-        fisheye_sensor_spec.position = [0, settings["sensor_height"], 0]
         for k in kw_args:
             setattr(fisheye_sensor_spec, k, kw_args[k])
         return fisheye_sensor_spec
 
-    if settings["fisheye_rgba_sensor"]:
-        fisheye_rgba_sensor_spec = create_fisheye_spec(uuid="fisheye_rgba_sensor")
-        sensor_specs.append(fisheye_rgba_sensor_spec)
-    if settings["fisheye_depth_sensor"]:
-        fisheye_depth_sensor_spec = create_fisheye_spec(
-            uuid="fisheye_depth_sensor",
-            sensor_type=habitat_sim.SensorType.DEPTH,
-            channels=1,
-        )
-        sensor_specs.append(fisheye_depth_sensor_spec)
-    if settings["fisheye_semantic_sensor"]:
-        fisheye_semantic_sensor_spec = create_fisheye_spec(
-            uuid="fisheye_semantic_sensor",
-            sensor_type=habitat_sim.SensorType.SEMANTIC,
-            channels=1,
-        )
-        sensor_specs.append(fisheye_semantic_sensor_spec)
-
     def create_equirect_spec(**kw_args):
         equirect_sensor_spec = habitat_sim.EquirectangularSensorSpec()
-        equirect_sensor_spec.uuid = "equirect_rgba_sensor"
-        equirect_sensor_spec.sensor_type = habitat_sim.SensorType.COLOR
-        equirect_sensor_spec.resolution = [settings["height"], settings["width"]]
-        equirect_sensor_spec.position = [0, settings["sensor_height"], 0]
         for k in kw_args:
             setattr(equirect_sensor_spec, k, kw_args[k])
         return equirect_sensor_spec
 
-    if settings["equirect_rgba_sensor"]:
-        equirect_rgba_sensor_spec = create_equirect_spec(uuid="equirect_rgba_sensor")
-        sensor_specs.append(equirect_rgba_sensor_spec)
-
-    if settings["equirect_depth_sensor"]:
-        equirect_depth_sensor_spec = create_equirect_spec(
-            uuid="equirect_depth_sensor",
-            sensor_type=habitat_sim.SensorType.DEPTH,
-            channels=1,
+    for sensor_cfg in settings["sensors"]:
+        sensor_cfg = {**default_sensor_settings, **sensor_cfg}
+        uuid = sensor_cfg["uuid"]
+
+        sensor_type = (
```

Comment:
Instead of pivoting off sensor name substrings, maybe we include "sensor_type" and "sensor_subtype" as fields in the sensor settings dict defaulting to color pinhole.

Reply:
sure, I think it might be okay to pivot off sensor name for the camera type (i.e. `fisheye`, `equirect`) unless we also pass in the camera spec type (i.e. `habitat_sim.FisheyeSensorDoubleSphereSpec`, `habitat_sim.EquirectangularSensorSpec`) in the sensor_settings_dict. But sensor_type and sensor_subtype will be updated to use the sensor settings dict